src/Routers/FilesRouter.js

```diff
@@ -10,7 +10,7 @@ export class FilesRouter {
 
   expressRouter(options = {}) {
     var router = express.Router();
-    router.get('/files/:appId/:filename', this.getHandler);
+    router.get('/files/:appId/:filename/:referencingClass?/:referencingKey?/:sessionToken?', this.getHandler);
```

Comment:
could we split the 2 endpoints for the sake a readability and ease of maintenance?

This way the endpoint that validates the auth calls the `getHandler` if everything is authorized. This will remove the clutter in the getHandler.